### DISCUSSION TITLE
New to version 2.3.6. and I have problems.

### DISCUSSION THREAD

**Discussion by WilliamFernandoC-P** [iqtree/iqtree2#336] [Q&A]
New to version 2.X.X.. and I have problems. 
Hi, for a long time I was using IQ-TREE 1.X.X, for ML phylogeny inference. I used the command “ iqtree -s <alignment> <parameters.... (especially -m MFP and Ultrafast bootstrap>”.  Today, I have a dataset of virus sequences that I had previously analyzed with version 1 but now with version 2 I get this error (for all sequences): ERROR: The sequence RSV_18_Consensus contains too many characters (953)....
I am trying to use sequences with different lengths <100, >1000, >1000... but with all I get the same thing. 

Can someone help me?, This is what happens

thanks

![Screenshot from 2024-10-23 20-51-16](https://github.com/user-attachments/assets/bf75342f-7858-4297-a178-e4adbde1555b)

**Comment by roblanf** [comment]
can you please upload your fasta file so we can take a look?

Rob

**Reply by roblanf** [reply]
@WilliamFernandoC-P, just pinging to see if you can upload your files. If you've managed to figure out this problem, please let us know so we can close off this discussion.

**Comment by thomaskf** [comment]
Are all the sequences inside the file of the same length?
The version you used was 2.0.7, which is quite old. Can you try the latest version from https://github.com/iqtree/iqtree2/releases and see if the problem still occurs?

Thomas

**Comment by roblanf** [comment]
closing this as no response from @WilliamFernandoC-P. 

If you do come back @WilliamFernandoC-P, feel free to post your files and we can take a look! 

Our concern (in case others read this thread) is that the sequences might be un-aligned.

**Reply by esomsen** [reply]
Hello, I recently encountered this same issue. I am attaching the log output. I am hesitant to share the source files, because not all sequences are public, but the input is a standard trimmed amino acid alignment (from mafft)
[cows_filt_trim_aa_aln.fas.log](https://github.com/user-attachments/files/22053517/cows_filt_trim_aa_aln.fas.log)

**Reply by roblanf** [reply]
@esomsen it's hard for us to help you without you sharing your input files. One option is to remove the unpublished stuff from your input and (assuming you have the same issue with the reduced file) post that one. Even better is if you can make a minimally reproducible example, e.g. by rerunning your pipeline on 10 or 20 sequences and posting those.

Either way, when MAFFT aligns sequences it should produce a file where all sequences are the same length, so the output is definitely surprising.

How long were you expecting your alignment to be? It's interesting that all the errors are 569 amino acids.

Rob

**Reply by esomsen** [reply]
Hi Rob, 

Excellent news! I was able to find the issue. After subsampling ~35 sequences a few times and having no issues with iqtree on those subsamples, I went through the original log file and found that the "too many characters" error was thrown for all but 10 sequences. I went through all the alignments after MAFFT from each postprocessing step (removing seqs with >10% Ns, trimming, translating). Apparently when the filtered and trimmed alignment was translated in MEGA, it truncated lengthy gaps at the end of some sequences, such that the 10 partial sequences at the top of the alignment were many aa shorter than the complete sequences.

Translating the alignment again and insuring no sequences were truncated resolved the issue. They are now happily running again. 

Thanks very much for your quick response which did help me figure out the issue. Happy to send you other output files if you should find them useful. 

Best,
Elizabeth
